File: two-sum/two-sum.js
var twoSum = function(nums, target) {
    const cache = {}
    for (let i = 0; i < nums.length; i++) {
        const val = nums[i], diff = target - val
        if (cache.hasOwnProperty(diff)) return [cache[diff], i]
        cache[val] = i
    }
    return []
};

File: two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    cache := map[int]int{}
    for i := 0; i < len(nums); i++ {
        diff := target - nums[i]
        value, exists := cache[diff]
        if exists {
            return []int{value, i}
        }
        cache[nums[i]] = i
    }
    return []int{}
}